plugins/plugin_utils/lookup.py

```diff
@@ -185,7 +199,7 @@ async def process_intermediate_path_part(self, intermediate_object_name):
                 self.set_new_filters_with_datacenter({"resource_pools": result})
                 return result
 
-        if self.object_type in ("host", "resource_pool"):
+        if self.object_type in ("vm", "host", "resource_pool"):
```

Comment:
Why did you added the "vm"? it's not clear 
If it is VM it will get inside the previous VM

Reply:
The function become not clear what each "if" statement comes to check and return 
can you refactoring this long function?

Reply:
The vm, host, and resource pool objects accept a cluster name as part of their filter spec. https://vdc-download.vmware.com/vmwb-repository/dcr-public/1cd28284-3b72-4885-9e31-d1c6d9e26686/71ef7304-a6c9-43b3-a3cd-868b2c236c81/doc/structures/com/vmware/vcenter/vm/svc.filter_spec-structure.html
So this change is fixing that

Reply:
I refactored the method and renamed it to `add_intermediate_path_part_to_filter_spec`. I also added some more comments and documentation, hopefully its clearer now